shared/src/test/scala/FutureTimeoutTests.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016-2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import org.scalatest._
+import cats.data.NonEmptyList
+import fetch._
+import fetch.implicits._
+
+class FutureTimeoutTests extends AsyncFlatSpec with Matchers with OptionValues with Inside with Inspectors {
+
+  implicit override def executionContext : ExecutionContext = ExecutionContext.Implicits.global
+
+  case class ArticleId(id: Int)
+  case class Article(id: Int, content: String) {
+    def author: Int = id + 1
+  }
+
+  def article(id: Int)(implicit DS: DataSource[ArticleId, Article]): Fetch[Article] =
+    Fetch(ArticleId(id))
+
+  // Simulate Thread.sleep for the purposes of making this test work in Scala.js
+
+  def sleep(ms: Long): Unit = {
+    println("start sleep")
+    val expTime = System.nanoTime + ms*1000000
+    while(System.nanoTime < expTime) {}
+    println("sleep finished")
+  }
```

Comment:
I am not sure if we can actually test the time out in ScalaJS.

I think that if you block like this, you block everything in JS, so the `TimerTask` cannot finish the promise with a `TimeoutExecption` (which is happening in the failing test on Travis).

We could revert back to using `Thread.sleep` and add this as a JVM only test, by moving it to `jvm/src/test/scala`, what do you think?